scripts/vipers/build_parent_sample.py

```diff
@@ -0,0 +1,150 @@
+import argparse
+import glob
+import os
+from multiprocessing import Pool
+import tarfile
+
+import h5py
+import healpy as hp
+import numpy as np
+import requests
+from astropy.io import fits
+from astropy.table import Table
+from tqdm import tqdm
+
+
+URL = "http://vipers.inaf.it/data/pdr2/spectra/"
+SURVEYS = ["VIPERS_W1_SPECTRA_1D_PDR2.tar.gz", "VIPERS_W4_SPECTRA_1D_PDR2.tar.gz"]
+
+SURVEY_SAVE_DIRS = ["vipers_w1", "vipers_w4"]
+HEADER_KEYS = ['ID', 'RA', 'DEC', 'REDSHIFT', 'REDFLAG', 'EXPTIME', 'NORM', 'MAG']
+
+
+def download_data(vipers_data_path: str = ''):
+    """Download the VIPERS data from the web and unpack it into the specified directory."""
+    # Create the output directory if it does not exist
+    if not os.path.exists(vipers_data_path):
+        os.makedirs(vipers_data_path)
+
+    # Download each file
+    for file in SURVEYS:
+        local_path = os.path.join(vipers_data_path, file)
+        subdirectory_path = os.path.join(vipers_data_path, file.replace(".tar.gz", ""))
+
+        # Create a subdirectory for each file
+        if not os.path.exists(subdirectory_path):
+            os.makedirs(subdirectory_path)
+
+        # Check if file needs to be downloaded
+        if not os.path.exists(local_path):
+            print(f"Downloading {file}...")
+            response = requests.get(URL + file, stream=True)
+            if response.status_code == 200:
+                with open(local_path, 'wb') as f:
+                    for chunk in response.iter_content(chunk_size=8192):
+                        f.write(chunk)
+            else:
+                print(f"Failed to download {file}. Status code: {response.status_code}")
+                continue
+
+        # Unpack the tar.gz file into its specific subdirectory
+        print(f"Unpacking into {subdirectory_path}...")
+        with tarfile.open(local_path, "r:gz") as tar:
+            tar.extractall(path=subdirectory_path)
+        print(f"Unpacked successfully!\n")
+
+        # Remove the tar files
+        os.remove(local_path)
+
+
+def extract_data(filename):
+    """Extract the contents of a tar file to a dictionary for each file"""
+    hdu = fits.open(filename)
+    header = hdu[1].header
+    data = hdu[1].data
+
+    results = {}
+
+    # Loop through the header keys and add them to the results dictionary
+    for key in HEADER_KEYS:
+        results[key] = float(header[key])
+    
+    # Add the spectrum data to the results dictionary
+    results['spectrum_flux'] = data['FLUXES'].astype(np.float32)
+    results['spectrum_wave'] = data['WAVES'].astype(np.float32)
+    results['spectrum_ivar'] = data['NOISE'].astype(np.float32)
+    results['spectrum_mask'] = data['MASK'].astype(np.float32)
```

Comment:
We will want to normalize these values somehow. What are the units of the fluxes/waves/noise/mask? How to move these to standardized units?